### ISSUE TITLE
Setup release workflow with GitHub Actions

### ISSUE THREAD

**Issue by ikysil** [cdi-unit/cdi-unit#189]
TODO:

- [x] Create release workflow
- [ ] Upload to GitHub release
- [x] Upload to Maven Central
- [x] Configure secrets

**Comment by ikysil** [comment]
#188 is the pre-requisite

**Comment by ikysil** [comment]
@BrynCooke Your assistance is required to set up the upload to Maven Central.

**Comment by BrynCooke** [comment]
Ok. It's been years since I did this. I think I can contact sonatype and get you added to the list of people that can do a release.

**Comment by ikysil** [comment]
I plan to use the process described at
https://docs.github.com/en/actions/guides/publishing-java-packages-with-maven#publishing-packages-to-the-maven-central-repository-and-github-packages

**Comment by BrynCooke** [comment]
Do you have a user id on Sonatype oss already? If not please sign up and let me know what it is.

**Comment by ikysil** [comment]
Do you mean the account to access https://issues.sonatype.org?

Yes, I have one - ikysil

I managed to find the ticket for initial setup https://issues.sonatype.org/browse/OSSRH-2949

**Comment by BrynCooke** [comment]
https://issues.sonatype.org/browse/OSSRH-63882

Can't believe that CDI unit is approx 8 years old.

**Comment by BrynCooke** [comment]
Looks like they granted you permissions already. Let me know if there's anything else you need.

**Comment by ikysil** [comment]
I've managed to release and upload 4.2.0-RC1 from my machine to the staging repository @ Sonatype.
I've used the `make-release.sh` script present in the repository to perform the release.

https://oss.sonatype.org/content/repositories/orgjgluecdi-unit-1003/

All the checks passed.


I think we should release 4.2.0 using the same process and investigate if it is possible to use GitHub workflow later.
@BrynCooke Any thoughts/suggestions?
